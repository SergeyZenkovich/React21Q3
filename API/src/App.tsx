import React, {useEffect, useState} from "react";
import {Switch, Route} from "react-router-dom";
import styled from "styled-components";

import HomePage from "./Components/HomePage/HomePage";
import AboutPage from "./Components/AboutPage/AboutPage";
import Header from "./Components/Header/Header";
import ErrorBlock from "./Components/ErrorComponent/ErrorComponent";
import DetailsPage from "./Components/DetailsPage/DetailsPage";
import {FetchObject} from "./interfaces/ComponentsInterfaces";

const MainComponent = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const App = (): JSX.Element => {
  const [activePage, setActivePage] = useState("Home");
  const [pageElements, setPageElements] = useState([] as FetchObject[]);

  return (
    <div>
      <Header activePage={activePage} setActivePage={setActivePage} />
      <MainComponent>
        <Switch>
          <Route exact path="/about" render={() => <AboutPage setActivePage={setActivePage} />} />
          <Route
            exact
            path="/details/:id"
            render={() => <DetailsPage setActivePage={setActivePage} />}
          />
          <Route
            exact
            path="/"
            render={() => (
              <HomePage setActivePage={setActivePage} setPageElements={setPageElements} />
            )}
          />
          <Route path="/*" render={() => <ErrorBlock />} />
        </Switch>
      </MainComponent>
    </div>
  );
};
export default App;
