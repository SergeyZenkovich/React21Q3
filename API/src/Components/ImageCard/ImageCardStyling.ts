import styled from "styled-components";
import device from "../../utils/sizes";

export const StyledCard = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 24%;
  @media ${device.mobileS} {
    min-width: 160px;
  }
  @media ${device.laptop} {
    min-width: 275px;
  }
  height: 50%;
  min-height: 200px;
  margin: 15px 5px;
  border: 1px solid #000;
  & img {
    width: 100%;
    height: 100%;
  }
`;
