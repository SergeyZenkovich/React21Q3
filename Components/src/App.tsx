import React from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const Cards = cardsData.map((el: CardInterface) => (
    <Card
      key={el.cardName}
      cardName={el.cardName}
      cardPrice={el.cardPrice}
      cardYear={el.cardYear}
      cardLikes={el.cardLikes}
      cardRating={el.cardRating}
      cardImg={el.cardImg}
    />
  ));
  return (
    <div>
      <Serach />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
