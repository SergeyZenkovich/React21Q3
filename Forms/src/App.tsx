import React, {useState} from "react";
import styled from "styled-components";
import Card from "./Components/Card/Card";
import Form from "./Components/Form/Form";
import Serach from "./Components/SearchBar/SearchBar";
import cardsData from "./Data";
import {CardInterface} from "./interfaces/ComponentsInterfaces";

const StyledCardsBlock = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const App = (): JSX.Element => {
  const [cards, setCards] = useState([] as CardInterface[]);
  const addCard = (newCard: CardInterface): void => {
    setCards([...cards, newCard]);
  };
  const Cards = cards.map((el: CardInterface) => (
    <Card
      key={el.cardProblem}
      cardProblem={el.cardProblem}
      cardDate={el.cardDate}
      cardPriority={el.cardPriority}
      cardisRequestNeeded={el.cardisRequestNeeded}
    />
  ));
  return (
    <div>
      <Form addCard={addCard} />
      <StyledCardsBlock>{Cards}</StyledCardsBlock>
    </div>
  );
};
export default App;
