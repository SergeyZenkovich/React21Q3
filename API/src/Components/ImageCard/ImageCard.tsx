import React from "react";
import styled from "styled-components";
import {FetchObject, ImageCardInterface} from "../../interfaces/ComponentsInterfaces";

import listImg from "../../assets/img/list.png";
import dateImg from "../../assets/img/calendar.png";
import likeImg from "../../assets/img/feedback.png";
import prodImg from "../../assets/img/prod.png";

import {StyledCard} from "./ImageCardStyling";

const ImageCard = ({id, color, likes, alt_description, user, urls}: FetchObject): JSX.Element => (
  <StyledCard>
    <img src={urls.regular} alt="" />
  </StyledCard>
);
export default ImageCard;
