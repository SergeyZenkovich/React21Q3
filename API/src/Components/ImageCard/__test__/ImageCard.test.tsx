import React, {useState} from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import {act} from "react-dom/test-utils";
import {HashRouter} from "react-router-dom";
import ImageCard from "../ImageCard";
import App from "../../../App";

const testObject = {
  id: "testId",
  urls: {
    full: "fullTest",
    regular: "regularTest",
    small: "smallTest",
    thumb: "thumbTest"
  }
};

it("renders without crashing", () => {
  const container = document.createElement("div");
  ReactDOM.render(
    <HashRouter>
      {" "}
      <ImageCard {...testObject}></ImageCard>
    </HashRouter>,
    container
  );

  expect(container.querySelector("img").alt).toBe("imageCard");
});

describe("Image Card Component", () => {
  const setFunction = jest.fn();
  let container;
  beforeEach(() => {
    container = renderer.create(
      <HashRouter>
        <ImageCard {...testObject}></ImageCard>
      </HashRouter>
    );
  });
  it("matches shapshot", () => {
    const tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
