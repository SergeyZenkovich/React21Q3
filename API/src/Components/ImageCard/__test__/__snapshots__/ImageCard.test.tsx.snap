// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Card Component matches shapshot 1`] = `
<div
  className="sc-bdnxRM fjwxUt"
>
  <a
    aria-current={null}
    href="#/details/testId"
    onClick={[Function]}
  >
    <img
      alt="imageCard"
      src="regularTest"
    />
  </a>
</div>
`;
